### PR TITLE
mosaic_cubes: pass on kwargs

### PR DESCRIPTION
**Pull request by keflavich**
The header determiner should accept kwargs so we can specify frames.

### CODE REVIEW

File: spectral_cube/cube_utils.py [hunk 1]
```diff
@@ -791,16 +792,16 @@ def combine_headers(header1, header2):
...
+    This function reprojects cubes onto a common grid and combines them to a single field.
 
     Parameters
     ----------
     cubes : iterable
-        Iterable list of SpectralCube objects to reproject and add together. 
+        Iterable list of SpectralCube objects to reproject and add together.
     spectral_block_size : int
-        Block size so that reproject does not run out of memory. 
+        Block size so that reproject does not run out of memory.
```
Comment (e-koch) [inline spectral_cube/cube_utils.py:805]: Add a docstring for `combine_header_kwargs` on passing these to `reproject.mosaicking.find_optimal_celestial_wcs`

Comment (e-koch) [reply spectral_cube/cube_utils.py:805]:
```suggestion
combine_header_kwargs : dict
     Keywords passed to `~reproject.mosaicking.find_optimal_celestial_wcs`.
```

**Review by e-koch — APPROVED** [review]
Looks good. Just add a docstring linking to reproject.

Link to check if the docs build doesn't handle it for reproject: `https://reproject.readthedocs.io/en/stable/api/reproject.mosaicking.find_optimal_celestial_wcs.html#reproject.mosaicking.find_optimal_celestial_wcs`

File: spectral_cube/cube_utils.py [hunk 2]
```diff
@@ -791,17 +792,18 @@ def combine_headers(header1, header2):
...
     ----------
     cubes : iterable
-        Iterable list of SpectralCube objects to reproject and add together. 
+        Iterable list of SpectralCube objects to reproject and add together.
     spectral_block_size : int
-        Block size so that reproject does not run out of memory. 
-
+        Block size so that reproject does not run out of memory.
+combine_header_kwargs : dict
+     Keywords passed to `~reproject.mosaicking.find_optimal_celestial_wcs`.
```
Comment (keflavich) [inline spectral_cube/cube_utils.py:806]:
```suggestion
    combine_header_kwargs : dict
         Keywords passed to `~reproject.mosaicking.find_optimal_celestial_wcs`.
```

**Pull request merged**